### DISCUSSION TITLE
Curiosity: why a lock around random? (retry.go jitter)

### DISCUSSION THREAD

**Discussion by mxmaxime** [go-resty/resty#632] [General]
Hi,
I was looking at the code of the jitter/backoff implementation, and I found something that peaked my curiosity.

[retry.go](https://github.com/go-resty/resty/blob/9943eabd71dea9e50698a7c24a1a83e10244de19/retry.go#L225)
```go
func randDuration(center time.Duration) time.Duration {
	rndMu.Lock()
	defer rndMu.Unlock()

	var ri = int64(center)
	var jitter = rnd.Int63n(ri)
	return time.Duration(math.Abs(float64(ri + jitter)))
}
```

Why did you wrap the rand in a lock? I guess it's because it needs to be thread-safe to use in != goroutines but still uses the methods on rand and not the top level functions of rand?

Thanks!
Have a nice one.

**Comment by jeevatkm** [comment]
@mxmaxime For thread-safe purposes when calling `rnd.Int63n(ri)`.
